luaa.h

```diff
@@ -31,6 +31,10 @@
 #include "common/lualib.h"
 #include "common/luaclass.h"
 
+#if !(501 <= LUA_VERSION_NUM && LUA_VERSION_NUM < 504)
+print("Awesome only supports Lua versions 5.1-5.3, please refer to https://awesomewm.org/apidoc/documentation/10-building-and-testing.md.html#Building")
```

Comment:
I think this should be 

```
fprintf(stderr, "Awesome only supports Lua versions 5.1-5.3, please refer to https://awesomewm.org/apidoc/documentation/10-building-and-testing.md.html#Building");
```

Reply:
I don't think it matters really. This function is called outside a function scope. The compiler will fail compilation no matter what this function is called. I called it `print` just to signal in some way that we want the user to read it.

Searching through StackOverflow I couldn't find a way to fail compilation in build time in C w/o doing some trick like this.
c11 has `static_assert` function which looks better, but I thought it's not worth it to impose requirement for c11 for awesome just to use static_assert

Reply:
`#error`?

Reply:
Yup, forgot about this one. According to its [documentation](https://gcc.gnu.org/onlinedocs/cpp/Diagnostics.html) this can work.

How about something like this:
```suggestion
#error "Awesome only supports Lua versions 5.1-5.3 and LuaJIT2, please refer to https://awesomewm.org/apidoc/documentation/10-building-and-testing.md.html#Building"
```